File: source/script/ScriptEngine.java
package script;

import entity.Entity;
import scene.Scene;

public class ScriptEngine {
    public ScriptEngine() {}

    public void callStarts(Scene scene) {
        for (Entity entity : scene) {
            entity.callScriptStarts();
        }
    }

    public void callUpdates(Scene scene) {
        for (Entity entity : scene) {
            entity.callScriptUpdates();
        }
    }
}

File: source/script/ScriptEngine.kt
package script

import scene.Scene

class ScriptEngine {
    fun callStarts(scene: Scene) {
        for (entity in scene) {
            entity.callScriptStarts()
        }
    }

    fun callUpdates(scene: Scene) {
        for (entity in scene) {
            entity.callScriptUpdates()
        }
    }
}